compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -2529,12 +2533,28 @@ LogicalResult initGPULaunchConfig(FunctionOpInterface funcOp) {
         break;
       }
     }
+
+    if (auto scatterOp = dyn_cast<IREE::LinalgExt::ScatterOp>(op)) {
+      auto indices = scatterOp.getIndices();
+      if (indices.getDefiningOp() == nullptr) {
```

Comment:
nit: llvm prefers casts to bool over explicit comparisons with nullptr unless we are dealing with custom nullable types or double-nullable types like `optional<Value>` etc.
```suggestion
      if (!indices.getDefiningOp()) {
```

Reply:
Thanks for the background